src/main/java/uk/gov/hmcts/divorce/document/content/DivorceApplicationSoleTemplateContent.java

```diff
@@ -142,13 +141,29 @@ private void setSolicitorDetails(Map<String, Object> templateContent, Applicant
         String solicitorName = applicant.getSolicitor().getName();
         String solicitorEmail = applicant.getSolicitor().getEmail();
         String solicitorFirmName = applicant.getSolicitor().getFirmName();
-        String solicitorAddress = applicant.getSolicitor().getAddress();
+        String solicitorAddress = applicant.getSolicitor().getAddress().replace("\n", "");
         boolean hasEnteredSolicitorDetails =
-            nonNull(solicitorName) || nonNull(solicitorEmail) || nonNull(solicitorFirmName) || nonNull(solicitorAddress);
+            !isNullOrEmpty(solicitorName)
+            || !isNullOrEmpty(solicitorEmail)
+            || !isNullOrEmpty(solicitorFirmName)
+            || !isNullOrEmpty(solicitorAddress);
         templateContent.put(APPLICANT_1_HAS_ENTERED_RESPONDENTS_SOLICITOR_DETAILS, hasEnteredSolicitorDetails);
-        templateContent.put(APPLICANT_2_SOLICITOR_NAME, solicitorName);
-        templateContent.put(APPLICANT_2_SOLICITOR_EMAIL, solicitorEmail);
-        templateContent.put(APPLICANT_2_SOLICITOR_FIRM_NAME, solicitorFirmName);
-        templateContent.put(APPLICANT_2_SOLICITOR_ADDRESS, solicitorAddress);
+        if (!isNullOrEmpty(solicitorName)) {
+            templateContent.put(APPLICANT_2_SOLICITOR_NAME, solicitorName);
+        }
+        if (!isNullOrEmpty(solicitorEmail)) {
+            templateContent.put(APPLICANT_2_SOLICITOR_EMAIL, solicitorEmail);
+        }
+        if (!isNullOrEmpty(solicitorFirmName)) {
+            templateContent.put(APPLICANT_2_SOLICITOR_FIRM_NAME, solicitorFirmName);
+        }
+        if (!isNullOrEmpty(solicitorAddress)) {
+            String addressCleanUp =
+                join("\n",
+                    Arrays.stream(applicant.getSolicitor().getAddress().split("\n"))
```

Comment:
the FE has different inputs for solicitor address so if the user only inputs line 1 and postcode for example, itll end up being `line1\n\n\n\npostcode` which doesnt not look great on the document hence the idea was to get rid of the extra new lines and have it look like `line1\npostcode`

Reply:
I was wondering if we need to do the same for the address for the applicants, as that may return the Solicitor address if they are represented

Reply:
or unless a CW just added multiple lines in the textarea